Data/samples/DBLogger/src/DBLogger.cpp

```diff
@@ -0,0 +1,337 @@
+			Option("constring", "s", "database connection string")
+				.required(true)
+				.repeatable(false)
+				.argument("cstr")
+				.callback(OptionCallback<DBLogger>(this, &DBLogger::handleDbConnectionString)));
+
+	}
+
+	void handleHelp(const std::string& name, const std::string& value)
+	{
+		_helpRequested = true;
+		displayHelp();
+		stopOptionsProcessing();
+	}
+
+	void handleLogDirectory(const std::string& name, const std::string& value)
+	{
+		if (value.empty())
+			throw Poco::Util::IncompatibleOptionsException("Expected directory name");
+		_directory = value;
+	}
+
+	void handleDbConnector(const std::string& name, const std::string& value)
+	{
+		if (value.empty())
+			throw Poco::Util::IncompatibleOptionsException("Expected database connector name");
+		_connector = value;
+	}
+
+	void handleDbConnectionString(const std::string& name, const std::string& value)
+	{
+		if (value.empty())
+			throw Poco::Util::IncompatibleOptionsException("Expected database connection string");
+		_connectionString = value;
+	}
+
+	void displayHelp()
+	{
+		HelpFormatter helpFormatter(options());
+		helpFormatter.setCommand(commandName());
+		helpFormatter.setUsage("OPTIONS");
+		helpFormatter.format(std::cout);
+	}
+
+	void onItemAdded(const Poco::DirectoryWatcher::DirectoryEvent& ev)
+	{
+		logger().trace("Added: %s", ev.item.path());
+		processFile(ev.item);
+	}
+
+	void onItemModified(const Poco::DirectoryWatcher::DirectoryEvent& ev)
+	{
+		logger().information("Modified: %s", ev.item.path());
+		processFile(ev.item);
+	}
+
+	void onItemMovedFrom(const Poco::DirectoryWatcher::DirectoryEvent& ev)
+	{
+		logger().trace("Moved from: %s", ev.item.path());
+		processFile(ev.item);
+	}
+
+	void onItemMovedTo(const Poco::DirectoryWatcher::DirectoryEvent& ev)
+	{
+		logger().trace("Moved to: %s", ev.item.path());
+	}
+
+	void onItemRemoved(const Poco::DirectoryWatcher::DirectoryEvent& ev)
+	{
+		logger().trace("Removed: %s", ev.item.path());
+	}
+
+	void processFile(const Poco::File& file)
+	{
+		// TODO: What to do with symbolic links?
+		// TODO: What to do if the file content is modified with correct content after it is added?
+		//       Ignore if empty and process on modified?
+
+		if (_dataSession == nullptr || !_dataSession->isGood())
+		{
+			return;
+		}
+
+		if (!file.isFile())
+		{
+			logger().trace("Not a file: %s", file.absolutePath());
+			return;
+		}
+		if (!_sqlNameGlob.match(file.absolutePath()))
+		{
+			logger().trace("Not an SQL file: %s", file.absolutePath());
+			return;
+		}
+		logger().debug("Will process: %s", file.absolutePath());
+
+		Poco::File f(file);
+		if (!f.exists())
+		{
+			logger().information("File does not exist: %s", file.absolutePath());
+			return;
+		}
+		try
+		{
+			Poco::FileInputStream is(f.absolutePath());
+			std::stringstream buffer;
+			buffer << is.rdbuf();
+			const auto& sql { buffer.str() };
+			if (!sql.empty())
+			{
+				auto& s = (*_dataSession);
+
+				s << sql, now;
+
+				s.isGood();
```

Comment:
no need for this. If sql succeeded, it's all fine. If it doesn't succeed, it will throw.

Reply:
This was left in by mistake.